website/docs/r/cdn_frontdoor_origin.html.markdown

```diff
@@ -0,0 +1,189 @@
+
+* `private_link` - (Optional) A `private_link` block as defined below.
+
+-> **Note:** Private Link requires that the CDN FrontDoor Profile this Origin is hosted within is using the SKU `Premium_AzureFrontDoor` - and that the `certificate_name_check_enabled` field above is set to `true`.
+
+* `weight` - (Optional) The weight of the origin in a given origin group for load balancing. Must be between `1` and `1000`. Defaults to `500`.
+
+---
+
+A `private_link` block supports the following:
+
+-> **Note:** At this time the Private Link Endpoint **must be approved manually** - for more information please see the [product documentation](https://docs.microsoft.com/azure/frontdoor/private-link).
+
+-> **Note:** To associate a Load Balancer with a CDN FrontDoor Origin via Private LInk you must stand up your own `azurerm_private_link_service` - and ensure that a `depends_on` exists on the `azurerm_cdn_frontdoor_origin` resource to ensure it's destroyed before the `azurerm_private_link_service` resource (e.g. `depends_on = [azurerm_private_link_service.example]`) due to the design of the CDN FrontDoor Service. 
+
+* `request_message` - (Optional) Specifies the request message that will be submitted to the `private_link_target_id` when requesting the private link endpoint connection. Values must be between `1` and `140` characters in length. Defaults to `Access request for CDN Frontdoor Private Link Origin`.
+
+* `target_type` - (Optional) Specifies the type of target for this Private Link Endpoint. Possible values are `blob`, `blob_secondary`, `web` and `sites`.
+
+-> **Note:** `target_type` cannot be specified when using a Load Balancer as an Origin.
+
+* `location` - (Required) Specifies the location where the Private Link resource should exist.
+
+~> **Note:** For performance reasons this value should match the location of the `private_link_target_id` resource.
+
+* `private_link_target_id` - (Required) The ID of the Azure Resource to connect to via the Private Link.
+
+---
+
+## Example HCL Configurations
+
+* [Private Link Origin with Storage Account Blob](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/examples/private-endpoint/cdn-frontdoor/storage-account-blob)
+* [Private Link Origin with Storage Account Static Web Site](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/examples/private-endpoint/cdn-frontdoor/storage-account-static-site)
+* [Private Link Origin with Linux Web Application](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/examples/private-endpoint/cdn-frontdoor/linux-web-app)
+* [Private Link Origin with Internal Load Balancer](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/examples/private-endpoint/private-link-service/cdn-frontdoor/load-balancer)
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported:
```

Comment:
is it possible to export the private endpoint connection resource id, when using Private Link? that would make approval much easier instead of trying to pull out the connection afterwards.